pkg/helm/io.go

```diff
@@ -28,28 +28,27 @@ func Load(yamlPath string) (ValuesMap, error) {
 	return values, nil
 }
 
-// FilterImages takes a ValuesMap and returns a map of images that
-// were found at the top level, or one level down with keys of
-// "image: "
-func FilterImages(values ValuesMap) map[string]string {
+// FilterImagesMaxLevel takes a ValuesMap and returns a map of images that
+// were found upto max level
+func FilterImagesMaxLevel(values ValuesMap, maxLevel int) map[string]string {
 	images := map[string]string{}
 
 	for k, v := range values {
-
-		// Match anything at the top level called "image: ..."
 		if k == "image" {
-			images[k] = v.(string)
+			imageUrl := v.(string)
+			images[imageUrl] = imageUrl
 		}
 
-		// Match anything at one level down i.e. "gateway.image: ..."
-		if c, ok := v.(map[interface{}]interface{}); ok && c != nil {
-			for kk, vv := range c {
-				if kk == "image" {
-					images[k] = vv.(string)
-				}
-			}
+		if c, ok := v.(ValuesMap); ok && maxLevel > 0 {
+			images = mergeMaps(images, FilterImagesMaxLevel(c, maxLevel-1))
```

Comment:
Interesting use of recursion. I had to think for a moment as to what you were doing here.

Reply:
Repeating code or use of something like stack to keep track of level felt a bit additional to me.